src/main/java/qupath/bioimageio/spec/BioimageIoSpec.java

```diff
@@ -1511,5 +1546,250 @@ public Processing.ProcessingMode deserialize(JsonElement json, Type typeOfT, Jso
+								oj.get("scale").getAsDouble()
+						);
+						case "channel" -> {
+							var namesJSON = oj.get("channel_names").getAsJsonArray();
+							List<String> names = new LinkedList<>();
+							for (JsonElement n: namesJSON) {
+								names.add(n.getAsString());
+							}
+							yield new ChannelAxis(
+									id, desc,
+									names
+							);
+						}
+						case "index" -> new IndexAxisBase(id, desc);
+						case "space" -> new SpaceAxisBase(
+								id, desc,
+								SpaceUnit.valueOf(oj.get("unit").getAsString().toUpperCase()),
+								oj.get("scale").getAsDouble()
+						);
+						case "batch" -> new BatchAxis(id, desc, oj.get("size"));
+						default -> {
+							logger.error("Unknown object {}", oj);
+							yield null;
+						}
+					};
+				}
+				return axes;
+			}
+			logger.error("Unknown JSON element {}", jsonElement);
+			return null;
+		}
+	}
+
+	public static abstract class Axis {
+	}
+
+	/**
+	 * Simple class to hold the old "bcyx" format from 0.4
+	 */
+	public static class CharAxis extends Axis {
+		private final char axis;
+
+		public CharAxis(char c) {
+			this.axis = c;
+		}
+	}
+
+	public static abstract class AxisBase extends Axis {
+		private String id;
+		private String description;
+		public AxisBase(String id, String description) {
+			this.id = id;
+			this.description = description;
+		}
+
+		public AxisBase(JsonElement id, JsonElement description) {
+			this(id == null ? "": id.getAsString(), description == null ? "": description.getAsString());
+		}
+	}
+
+	public static class BatchAxis extends AxisBase {
+		private int size;
+		private boolean concatenable = true;
+		public BatchAxis(JsonElement id, JsonElement description, JsonElement size) {
+			super(id, description);
+			int s = 1;
+			if (size != null) {
+				s = size.getAsInt();
+			}
+			this.size = s;
+		}
+	}
+
+	public static class ChannelAxis extends AxisBase {
+		private List<String> channel_names;
+
+		public ChannelAxis(JsonElement id, JsonElement description, List<String> channel_names) {
+			super(id, description);
+			this.channel_names = channel_names;
+		}
+
+		public int size() {
+			return channel_names.size();
+		}
+	}
+
+	public static class IndexAxisBase extends AxisBase {
+		private double scale = 1.0;
+		private String unit = null;
+
+		public IndexAxisBase(JsonElement id, JsonElement description) {
+			super(id, description);
+		}
+	}
+
+	public static class IndexInputAxis extends IndexAxisBase {
+		private boolean concatenable = false;
+
+		public IndexInputAxis(JsonElement id, JsonElement description) {
+			super(id, description);
+		}
+	}
+
+	public static class IndexOutputAxis extends IndexAxisBase {
+		private Size size;
+
+		public IndexOutputAxis(JsonElement id, JsonElement description) {
+			super(id, description);
+		}
+	}
+
+	public static class Size {
+		// todo: handle ints or SizeReference of DataDependentSize...???
+	}
+
+	public static class DataDependentSize extends Size {
```

Comment:
Does this have other be a public class for deserialisation?
Feels like if the class is public then the fields should have getters - otherwise the class needn't be public.